src/main/scala/microc/cli/Actions.scala

```diff
@@ -169,3 +182,72 @@ case class TypeAction(file: File,
     }
   }
 }
+
+case class CfgAction(file: File, parserName: String = Parser.DefaultParserName)
+  extends Action with ParsingAction {
+  override def run(): Int = {
+    val source = readInput(file)
+    val reporter = new Reporter(source, Some(file.getPath))
+
+    try {
+      val program = parser.parseProgram(source)
+      val (declarations, fieldNames) = new SemanticAnalysis().analyze(program)
+      // TODO type analysis too
+      val normalized = AstNormalizer.normalize(program)
+      val cfg = Cfg.convert(normalized)
+      println(cfg.toDot)
+      0
+    } catch {
+      case e: ProgramException =>
+        System.err.println(e.format(reporter))
+        1
+    }
```

Comment:
The _parse, analyze, normalize, and catch exceptions_ pattern appears over and over and should be abstracted into some sort of a pipeline.